feat: Add support for `vite serve`

mejo- opened the pull request:
Fixes: #6732

### 🚧 TODO

- [x] Add support for a nginx config drop-in to nextcloud-docker-dev

### 🏁 Checklist

- [x] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [x] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [x] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

max-nextcloud commented:
Thanks a lot for looking into this. :heart:  I'm super happy this works.

I have two suggestions:
* use a non-standart port and add it to the vite config. Then we can hardcode the same port in nextcloud-docker-dev for text. I suggest 7347 😉.
* I will look into making use of `vite/@client` instead of the middleware. It looks like that rewrites the module urls inside the browser and could replace the middleware. See step 2 of https://v5.vite.dev/guide/backend-integration.html

max-nextcloud commented:
I tried to follow the instructions.
It proved a bit challenging to get the changes to nginx applied. Whenever i restarted the proxy container my changes were gone. So I tried to add the locations and then ran `/etc/init.d/nginx restart`. This printed some errors about not being able to bind but then at least my config would have an effect.

However now I am seeing this in the proxy logs:
```
proxy-1      | nginx.1     | 2024/12/11 09:34:19 [error] 63#63: *9 connect() failed (101: Network is unreachable) while connecting to upstream, client: 192.168.23.1, server: nextcloud.local, request: "GET /apps/text/js/text-init.mjs HTTP/1.1", upstream: "http://192.168.15.1:5173/apps/text/js/text-init.mjs", host: "nextcloud.local"
```
Maybe the ip addresses are different for me?

mejo- commented:
Thanks for testing it out @max-nextcloud. I added support for a nginx config drop-in to nextcloud-docker-dev in the meantime: https://github.com/juliusknorr/nextcloud-docker-dev/pull/367

This approach is also the reason why I'd keep the default port of `vite serve`: we won't add the configuration for forwarding Text request to `vite serve` to nextcloud-docker-dev anyway.  But in future it's as simple as running `npm run serve` in Text repo and adding the small snippet to `data/nginx/vhost.d/nextcloud.local` in nextcloud-docker-dev and that's it.

> Maybe the ip addresses are different for me?

Yes, that's adjusted now as well. The PR to `nextcloud-docker-dev` adds `host.docker.internal` to the proxy container, so no longer needed to fiddle around with IPs.

mejo- commented:
Unfortunately the approach to make vite serve text at `/` and rewrite the URLs accordingly didn't work out well. I reverted to the earlier approach and adjusted the docs. This PR is ready to review now.

juliusknorr commented on README.md:
````diff
@@ -40,6 +40,26 @@ Follow its development setup and then continue here.
+   When using [nextcloud-docker-dev](https://github.com/juliusknorr/nextcloud-docker-dev), add the following snippet to `data/nginx/vhost.d/nextcloud.local` and restart the proxy container. You might have to replace `/apps/text` with e.g. `/apps-extra/text` depending on where the text app resides in your dev setup.
+   ```
+   location /apps/text/ {
+       proxy_pass http://host.docker.internal:5173/apps/text;
````
```
proxy_pass http://host.docker.internal:5173/apps/text/;
```

I needed this traling slash to make it work 🤷
